tests/jballerina-unit-test/src/test/resources/test-src/expressions/access/field_access_negative.bal

```diff
@@ -376,3 +376,32 @@ function testInvalidBoundMethodAccessWithRemoteMethod(ServiceClass a,
 
     _ = e.ser.fn;
 }
+
+class ManagerR {
+    function func(int i) returns int => i;
+}
+
+class CompanyR {
+    function func(int i) returns string => (i+1).toString();
+}
+
+class EmployeeR {
+    function func(int i) returns int => i;
+}
+
+function testAccessingMethodOnUnionObjectTypeNegative() {
+    ManagerR|CompanyR ob1 = new CompanyR();
+    _ = ob1.func();
+
+    (function (int i) returns int)|function (int i) returns string func1 = ob1.func;
+    _ = func1(1);
+
+    var func2 = ob1.func;
+    _ = func2(1);
+
+    ManagerR|EmployeeR ob2 = new EmployeeR();
+    _ = ob2.func();
+
+    var func3 = ob1.func;
+    _ = func3(1);
```

Comment:
These should go in a separate test file IMO. E.g., with tests for method calls, since these are no longer field access.

Reply:
Updated 79d7ed91adf3807fa51e7ae00800fa937d4ea30a